brave/brave-core
# Add NFT gallery P3A metrics

Resolves https://github.com/brave/brave-browser/issues/32315



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

0. Start with fresh profile, setup the wallet, ensure no metrics exist.
1. View the NFT gallery, ensure the NFT count metric is 0, new user metric is 1.
2. When prompted to enable NFT discovery, do so. ensure discovery metric is set to 1.
3. Add an NFT to the gallery, ensure the NFT count metric is 1.
4. Ensure that the count and new user metrics have been sent, advance the system time by one week, restart browser. ensure that the metrics no longer exist
5. View the NFT gallery again. Ensure that the NFT count is set to 1, and the new user metric does not exist
6. Disable NFT discovery, ensure that the discovery metric is set to 0
